presto-orc/src/main/java/com/facebook/presto/orc/OrcRecordReader.java

```diff
@@ -338,6 +340,12 @@ public void close()
             List<ColumnStatistics> columnStatistics = fileStatisticsValidation.get().build();
             writeValidation.get().validateFileStatistics(orcDataSource.getId(), columnStatistics);
         }
+
+        for (StreamReader column : streamReaders) {
```

Comment:
This `close()` method needs some refactoring to make sure that all the stream readers are guaranteed to be closed. For example, what happens if  `orcDataSource.close()` throws above, or when validations above fail? You can use `com.google.common.io.Closer`, there are many examples of how to use it.